### PR TITLE
Support for clone a persistent disk

### PR DESCRIPTION
**Pull request by AarshDhokai**
Fixing the issue : [(t-g) Support for GCP Terraform provider to Clone a Persistent Disk ](https://b.corp.google.com/issues/228193793)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `source_disk` field to `google_compute_disk` and `google_compute_region_disk` resource
```

### CODE REVIEW

**Comment by mitj04** [comment]
LGTM

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -3507,6 +3507,25 @@ objects:
...
+        description: |
+          The source disk used to create this disk. You can provide this as a partial or full URL to the resource.
+          For example, the following are valid values:
+
+          * https://www.googleapis.com/compute/v1/projects/project/zones/zone/disks/disk
+          * https://www.googleapis.com/compute/v1/projects/project/regions/region/disks/disk
+          * projects/project/zones/zone/disks/disk
+          * projects/project/regions/region/disks/disk
+          * zones/zone/disks/disk
+          * regions/region/disks/disk
```
Comment (c2thorn) [inline mmv1/products/compute/api.yaml:3521]: For better formatting, can we surround the variable segments with {}? like `projects/{project}/zones/{zone}/disks/{disk}` for both versions of the field

**Review by c2thorn — CHANGES_REQUESTED** [review]
Very nice PR, I only have 1 minor request with formatting. Logic looks great

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (AarshDhokai) [reply mmv1/products/compute/api.yaml]: Done.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**